### PR TITLE
Use "denylist" in api-resources.py

### PR DESCRIPTION
**Pull request by tbarrella**
As part of https://github.com/istio/istio/issues/25381

It looks like values are never passed to override the defaults, making this seem safe

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Review by clarketm — APPROVED** [review]

File: prow/api-resources.py [hunk 1]
```diff
@@ -109,18 +109,18 @@ def main(delimiter, format, group_blacklist, kind_blacklist):
         "--format", type=str, default="%s", help="format string for each api resource."
     )
     parser.add_argument(
-        "--group-blacklist",
+        "--group-denylist",
```
Comment (cjwagner) [inline prow/api-resources.py:112]: Aren't these breaking changes? I would have expected both flags to be supported temporarily or for the uses of the script to be updated in this PR as well.

Comment (tbarrella) [reply prow/api-resources.py:112]: I couldn't find these flags being used in any Istio repo; it seems like only the default value is ever used

**Review by cjwagner — APPROVED** [review]
/hold
Ah I see in the PR body that you say the flags are never used.

File: prow/api-resources.py (hunk 1 above)
Comment (tbarrella) [reply prow/api-resources.py:112]: See for example #2551

**Comment by tbarrella** [comment]
/hold cancel
Merging since @clarketm approved and has been the only previous contributor to the file

File: prow/api-resources.py (hunk 1 above)
Comment (clarketm) [reply prow/api-resources.py:112]: Yes, I added these flags to make these attributes configurable (for any extended usage) but made is so we are using the _default_ values in this repo. And AFAIK this is not used anywhere else. So this should not be breaking for Istio.

**Pull request merged**